gpiozero/output_devices.py

```diff
@@ -1292,6 +1292,127 @@ def stop(self):
+        :data:`None` a :exc:`GPIODeviceError` will be raised.
+
+    :type forward: int or str
+    :param forward:
+        The GPIO pin that the forward input of the motor driver chip is
+        connected to. See :ref:`pin-numbering` for valid pin numbers. If this
+        is :data:`None` a :exc:`GPIODeviceError` will be raised.
+
+    :type backward: int or str
+    :param backward:
+        The GPIO pin that the backward input of the motor driver chip is
+        connected to. See :ref:`pin-numbering` for valid pin numbers. If this
+        is :data:`None` a :exc:`GPIODeviceError` will be raised.
+
+    :type pin_factory: Factory or None
+    :param pin_factory:
+        See :doc:`api_pins` for more information (this is an advanced feature
+        which most users can ignore).
+    """
+    def __init__(self, speed, forward, backward, *, pin_factory=None):
+        super().__init__(
+            speed_device=PWMOutputDevice(speed, pin_factory=pin_factory),
+            forward_device=DigitalOutputDevice(forward, pin_factory=pin_factory),
+            backward_device=DigitalOutputDevice(backward, pin_factory=pin_factory),
+            _order=('speed_device', 'forward_device', 'backward_device'),
+            pin_factory=pin_factory)
+
+    @property
+    def value(self):
+        """
+        Represents the speed of the motor as a floating point value between -1
+        (full speed backward) and 1 (full speed forward), with 0 representing
+        stopped.
+        """
+        if self.forward_device.value and not self.backward_device.value:
+            return self.speed_device.value
+        if self.backward_device.value and not self.forward_device.value:
+            return -self.speed_device.value
+        return 0
+
+    @value.setter
+    def value(self, value):
+        if not -1 <= value <= 1:
+            raise OutputDeviceBadValue("Motor value must be between -1 and 1")
+        if value > 0:
+            try:
+                self.forward(value)
+            except ValueError as e:
+                raise OutputDeviceBadValue(e)
+        elif value < 0:
+            try:
+                self.backward(-value)
+            except ValueError as e:
+                raise OutputDeviceBadValue(e)
+        else:
+            self.stop()
+
+    @property
+    def is_active(self):
+        """
+        Returns :data:`True` if the motor is currently running and
+        :data:`False` otherwise.
+        """
+        return self.value != 0
+
+    def forward(self, speed=1):
+        """
+        Drive the motor forwards.
+
+        :param float speed:
+            The speed at which the motor should turn. Can be any value between
+            0 (stopped) and the default 1 (maximum speed).
+        """
+        if not 0 <= speed <= 1:
+            raise ValueError('forward speed must be between 0 and 1')
+        self.backward_device.off()
+        self.forward_device.on()
+        self.speed_device.value = speed
```

Comment:
As a safety-measure, I wonder if this should _first_ set `self.speed_device.value` to `0` before turning the `forward_device` on? I.e. if you do
```python3
my_speed_motor.backward(1)
time.sleep(1)
my_speed_motor.forward(0.2)
```
you probably **don't** want a momentary blip where the motor is running forwards at full speed?